elastic/kibana
# [Search Sessions] omit searchSessionId from the initialState, explicitly pause refreshInterval in restoreState

## Summary

Closes https://github.com/elastic/kibana/issues/88567 https://github.com/elastic/kibana/issues/88568

This pr closes two-issue around restoring a search session in Discover and Dashboard:

### 1. `searchSessionId` should be omitted in `initialState` 

We save two states into a search session saved object: 
* `restoreState`
*  `initialState`
These states are used when navigating from management to an App. By default `restoreState` is used to restore a session. 
We use `initialState` when we know there is no session to restore, for example, in case the search session is in the `ERROR` or `EXPIRED` state. `initialState` should not contain `searchSessionId`, otherwise search requests will error out. 

#### How to test

When saving a session there should be no `searchSessionId` inside `initialState` in a request save request payload.

To test end-to-end needs https://github.com/elastic/kibana/pull/81707. In case the search is is in the `ERROR` or `EXPIRED` state, navigating to it should just restart a search.

### 2. refresh interval should be explicitly `paused` when generating `restoreState` for dashboard. 

If there is no `refreshInterval` in the URL dashboard picks up whatever is saved in the dashboard SO.
